OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/string/interpretation/InterpretationHandler.scala

```diff
@@ -0,0 +1,146 @@
+                InterpretationHandler.getEntity,
+                StringFlowFunctionProperty.ub,
+                Set(state.tacDependee),
+                continuation(state)
+            )
+        } else if (tacaiEOptP.ub.tac.isEmpty) {
+            // No TAC available, e.g., because the method has no body
+            StringInterpreter.computeFinalResult(StringFlowFunctionProperty.constForAll(StringTreeNode.lb))
+        } else {
+            processStatementForState
+        }
+    }
+
+    private def continuation(state: InterpretationState)(eps: SomeEPS): ProperPropertyComputationResult = {
+        eps match {
+            case FinalEP(_, _) if eps.pk.equals(TACAI.key) =>
+                state.tacDependee = eps.asInstanceOf[FinalEP[Method, TACAI]]
+                processStatementForState(state)
+
+            case _ =>
+                InterimResult.forUB(
+                    InterpretationHandler.getEntity(state),
+                    StringFlowFunctionProperty.ub,
+                    Set(state.tacDependee),
+                    continuation(state)
+                )
+        }
+    }
+
+    private def processStatementForState(implicit state: InterpretationState): ProperPropertyComputationResult = {
+        val defSiteOpt = valueOriginOfPC(state.pc, state.tac.pcToIndex);
+        if (defSiteOpt.isEmpty) {
+            throw new IllegalArgumentException(s"Obtained a pc that does not represent a definition site: ${state.pc}")
+        }
+
+        processStatement(state)(state.tac.stmts(defSiteOpt.get))
+    }
+
+    protected def processStatement(
+        implicit state: InterpretationState
+    ): PartialFunction[Stmt[V], ProperPropertyComputationResult]
+}
+
+object InterpretationHandler {
+
+    final val SoundnessModeConfigKey = "org.opalj.fpcf.analyses.string.InterpretationHandler.highSoundness"
+
+    def getEntity(implicit state: InterpretationState): MethodPC = MethodPC(state.pc, state.dm)
+
+    /**
+     * This function checks whether a given type is a supported primitive type. Supported currently
+     * means short, int, float, or double.
+     */
+    private def isSupportedPrimitiveNumberType(typeName: String): Boolean =
+        typeName == "short" || typeName == "int" || typeName == "float" || typeName == "double"
+
+    /**
+     * Checks whether a given type, identified by its string representation, is supported by the
+     * string analysis. That means, if this function returns `true`, a value, which is of type
+     * `typeName` may be approximated by the string analysis better than just the lower bound.
+     *
+     * @param typeName The name of the type to check. May either be the name of a primitive type or
+     *                 a fully-qualified class name (dot-separated).
+     * @return Returns `true`, if `typeName` is an element in [char, short, int, float, double,
+     *         java.lang.String] and `false` otherwise.
+     */
+    def isSupportedType(typeName: String): Boolean =
```

Comment:
Why is this method based on the toJava representation and string comparisons, not directly on OPAL `Type`s?